### PR TITLE
Svcd-Write Function 

### PR DESCRIPTION
**Pull request by AparnaDhinakaran**

### CODE REVIEW

File: natlib/svcd.c [hunk 1]
```diff
@@ -185,3 +196,208 @@ static int svcd_init( lua_State *L )
...
+    // REGISTER CLUA FUNCTIONS IN GLOBAL SPACE
+    // lua_pushcfunction(L, svcd_write_invoke_later);
+    // lua_setglobal(L, "svcd_write_invoke_later");
+    // lua_pushcfunction(L, anon_function);
+    // lua_setglobal(L, "anon_function");
+    // lua_pushcfunction(L, svcd_callback_handler);
+    // lua_setglobal(L, "svcd_callback_handler");
+    // END REGISTER
+
+    printf("Sexy write!\n");
```
Comment (pnigam) [inline natlib/svcd.c:281]: Can you remove the debugging print statements (Holy Cow)

**Comment by pnigam** [comment]
It seems like you have added additional files other than just the svcd file. To make this cleaner, please just include the svcd file. 

Holy Cow

**Comment by jenniferdai** [comment]
```
        lua_settable(L, -3);
        printf("Appending svcid 0x%x\n", svcid);
        //attrid
        lua_pushnumber(L, 2);
        lua_pushnumber(L, attrid);
        lua_settable(L, -3);
```

Why do you settable(L, -3) twice?

-birdpeople

**Comment by mccormickt12** [comment]
Im not sure if it matters but it might be cleaner to get svcid and the other args as uint16's like they are stored as instead of converting and pushing them as ints

--TBD

**Comment by NickFirmani** [comment]
You'll want to rebase your commits to remove the extra files, so that you only commit svcd

**Comment by leonardt** [comment]
If you don't want to mess around with trickier git commands like rebase and cherry pick you can use the method i described [here](https://github.com/UCB-IoET/ioet_contrib/pull/32#issuecomment-76583449) to manually pull in your new svcd.c from a fresh branch off of the current master but you will lose the commit history.

File: natlib/svcd.c (hunk 1 above)
Comment (jochemvang) [inline natlib/svcd.c:281]: I would like to echo the sentiments of Holy Cow. All these print statements are making your code hard to read efficiently. Feel free to use them for testing your code, but remove them when you push it here. 

Best, 
- PB for your Jelly or, The Perfect Storm

File: natlib/svcd.c [hunk 2]
```diff
@@ -185,3 +196,208 @@ static int svcd_init( lua_State *L )
...
+        ivkid = 0;
+    }
+
+    lua_settop(L, numargs);
+
+    printf("Updating ikvid to %i\n", ivkid);
+    // setting ivkid
+    lua_getglobal(L, "SVCD");
+    lua_pushnumber(L, ivkid);
+    lua_setfield(L, -2, "ikvid");
```
Comment (mschuldt) [inline natlib/svcd.c:315]: This chunk of code to increment ikvid exists twice in svcd.c, perhaps a they should both be replaced with a next_ikvid function
(Running With Scissors)

Comment (Michaelhobo) [inline natlib/svcd.c:315]: I can't seem to find the second instance of incrementing ivkid. What line is it on?

File: natlib/svcd.c (hunk 1 above)
Comment (Michaelhobo) [inline natlib/svcd.c:281]: Duly  noted.

**Comment by immesys** [comment]
This has been superseded by #39

**Pull request closed without merging**